File: uvaProblems/11729 Commando War.py
# Link: https://uva.onlinejudge.org/external/117/11729.pdf

def get_minimum_time(soldiers):
    res = total_brief_times = 0
    for brief_time, execution_time in soldiers:
        total_brief_times += brief_time
        res = max(res, total_brief_times + execution_time)
    return res

counter = 1
while True:
    test_cases = input()
    if test_cases == "0":
        break
    soldiers = []
    
    for i in range(int(test_cases)):
        brief_time, execution_time = [int(x) for x in input().split()]
        soldiers.append((brief_time, execution_time))

    soldiers.sort(key=lambda x: -x[1]) # Sort by descending time of execution time.
    print('Case {}: {}'.format(counter, get_minimum_time(soldiers)))
    counter += 1

File: uvaProblems/11729_Commando_War.java
// Runtime error :( but logic seems right..
// Still debugging but feel free to take a look at the Python version since it is Accepted (AC).

import java.io.*;
import java.util.*;
import java.util.stream.*;

class Soldier implements Comparable<Soldier> {
    int timeToBrief;
    int timeToExecute;
    
    Soldier(int b, int e) {
        timeToBrief   = b;
        timeToExecute = e;
    }
    
    @Override
    public int compareTo(Soldier other) {
        // Sort by descending order.
        return other.timeToExecute - this.timeToExecute;
    }
    
    @Override
    public String toString() {
        return "(" + timeToBrief + "," + timeToExecute + ")";
    }
}

class Main {
    
    private int getMinimumTime(List<Soldier> soldiers) {
        int sum = 0;
        int ans = 0;
        for(Soldier s: soldiers) {
            sum += s.timeToBrief;
            ans = Math.max(ans, sum + s.timeToExecute);
        }
        return ans;
    }
    
    public static void main (String[] args) throws IOException {
        Main m = new Main();
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        int counter = 1;
        while (true) {
            int testCases = Integer.parseInt(bf.readLine());
            if(testCases == 0) {
                break;
            }
            List<Soldier> soldiers = new ArrayList<>();
            for(int i = 0; i < testCases; i++) {
                int[] nums = Arrays.stream(bf.readLine().split(" "))
                                          .mapToInt(Integer::parseInt)
                                          .toArray();
                soldiers.add(new Soldier(nums[0], nums[1]));
            }
            Collections.sort(soldiers);
            System.out.println("Case " + counter + ": " + m.getMinimumTime(soldiers));
            counter += 1;
        }
    }
}
